src/st.py

```diff
@@ -11,5 +12,138 @@ def main():
           f"in genome {args.genome.name}")
 
 
-if __name__ == '__main__':
-    main()
+
+# Index does not work
+# Don't mind all the broken code 
+def constructTreeNaive(x: str):
+    x += "$"
+    n = len(x)
+
+    firstLeaf = Node(None, (1, n), 0)
```

Comment:
Doesn't the first leaf go from index 0? `Node(None, (0, n), 0)`?

Reply:
The invariant is that every node has startindex of the substring on the edge minus one. 
The reason for this is that the first character is already in the dictionary, that linked the parent to the node, so it is not needed.

Reply:
Okay. Makes sense